aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/SyncProcessor.java

```diff
@@ -74,6 +84,11 @@ private SyncProcessor(
         this.appSync = Objects.requireNonNull(appSync);
         this.merger = Objects.requireNonNull(merger);
         this.dataStoreConfigurationProvider = dataStoreConfigurationProvider;
+        this.modelNames = Observable.fromIterable(modelProvider.models())
+                                        .map(m -> m.getSimpleName())
+                                        .toList()
+                                        .blockingGet()
+                                        .toArray(new String[0]);
```

Comment:
See other comments about `ForEach`. This one returns `String[]` though instead of `List<String>`. I guess we'd have to standardize. Oh, or like `ForEach.inArray`, `ForEach.inList`, etc. Could do multiple collection types

Reply:
I changed the input parameter to be `Collection<T>` which covers a majority of cases and also added a `inArray` method in case someone needs it.